File: app/src/main/java/com/siju/acexplorer/storage/model/task/CustomKey.java
package com.siju.acexplorer.storage.model.task;

import com.bumptech.glide.load.Key;

import java.security.MessageDigest;


class CustomKey implements Key {

    private String path;

    CustomKey(String path) {
        this.path = path;
    }

    @Override
    public void updateDiskCacheKey(MessageDigest messageDigest) {

    }

    @Override
    public boolean equals(Object obj) {
        if (obj instanceof String) {
            return path.equals(obj);
        }
        return super.equals(obj);
    }

    @Override
    public int hashCode() {
        return super.hashCode();
    }
}

File: common/src/main/java/com/siju/acexplorer/common/customglide/CustomKey.kt
package com.siju.acexplorer.common.customglide

import com.bumptech.glide.load.Key
import java.security.MessageDigest

internal class CustomKey(private val path: String) : Key {
    override fun updateDiskCacheKey(messageDigest: MessageDigest) {}
    override fun equals(other: Any?): Boolean {
        return if (other is String) {
            path == other
        }
        else super.equals(other)
    }

    override fun hashCode(): Int {
        return path.hashCode()
    }
}